modules/ding_user/ding_user.module

```diff
@@ -593,7 +504,7 @@ function ding_user_provider_profile($account) {
  *
  * @todo: Implement hook in the providers that returns the name.
  *
- * @return entity
+ * @return ProfileType[]|bool
  *   The profile type entity or FALSE if not found.
```

Comment:
There is a mismatch between return type and comment here. Do we return one `ProfileType` object or an array of `ProfileType` objects?

Reply:
This is where it becomes a little stange. The documentation is correct in that the `profile2_type_load()` function will return an array if the parameter is not given. 

`Depending whether $type isset, an array of profile types or a single one.` 

So it will always not be an array in this case, but it I remove the [] the IDE will complain.

Reply:
I do not think we should introduce obvious errors in our code because of an IDE and poorly documented third party code. I suggest that we change the return type to `ProfileType|bool` and try one of the following:

1. Ignore the IDE warning
2. Add a type hint: `/* @var ProfileType|FALSE $profile_type */`
3. Add code which explicitly insures the value of `$profile_type`: `$profile_type = (is_array($profile_type)) ? array_shift($profile_type) : $profile_type`

Reply:
Fixed